### ISSUE TITLE
Re-stickify sidebar

### ISSUE THREAD

**Issue by hbillings** [18F/cms-hitech-apd#1337]
The sidenav lost its sticky with the redesign, partly because it seems like the nav no longer fits on some screens. We should re-stick it and address the design concerns as needed.

### This task is done when...
 - [x] We have a decision on how the nav should look and function. (For example, we've adjusted the padding to make things fit on a single screen, or we've made the nav wide enough to accommodate longer text on a single line.)
 - [x] The sidebar sticks.
 - [x] The stickiness works across all browsers.

**Comment by jenniferthibault** [comment]
@jameshupp before I change the scaffolding, is there anything we could do to shorten the parent and child-level titles through better content design? I'm not sure how recently this was visited.

**Comment by jameshupp** [comment]
I could use some constraints.

* How much shorter is useful? All on one line, fewer lines wherever they may come from, just shorter please who cares about lines, depends on which level of title it is how much shorter it should be?
* What is the goal or direction you're trying to go in here? Or if this could go multiple directions, what are the options, even loosely?

There probably are some changes we could make, but I want to get a sense there to see if what feels realistic in a short-ish timeframe is going to be helpful.

**Comment by jenniferthibault** [comment]
FAIR! Yes good point.

So. I don't have the rules, but I can tell you what's driving this need and maybe that helps us get to the rules together.

The primary in-page navigation is that side nav. We want it to be sticky so that it travels with the user down the page. Right now, the nav is taller in height than many browser windows, so you can't actually access any of the nav items at the bottom of the list.

When any accordion panel is expanded, you can't get to nav items at the bottom of the list until other panels above it are closed.

There are so many sections that this will never be perfect, and we'll always have this problem as long as the app is one long page instead of multiple pages. But it seems like the best thing we can do to make it as short as we can is to keep section titles (as they appear in the nav) as few lines as possible. One is GREAT! Two, less great. Three: really not great.

Glad to pair up if that helps!

**Comment by jenniferthibault** [comment]
Here are some additional tactics that could help:
1. Use just the state name and icon in the top left corner. Drop `HITECH APD`  to make more space.
Hypothesis: By now the user will know where they are and what they're doing.
2. We banish "Export" back into the nav as a final step (where users can get further instructions for what's next) instead of as a button available from everywhere: https://github.com/18F/cms-hitech-apd/issues/1321 
3. We keep working on the assumption that users need to be able to save their progress at any point in the process for peace of mind. We try using a fixed position "Save" button anchored to the bottom right corner of a user's screen, outside the grid. For screens 1024 and larger, it won't overlay any content. For screens smaller, it would overlay content. So it's a little imperfect, but might be worth trying for now.

![1_State Profile Contact](https://user-images.githubusercontent.com/11636908/54689439-ea4f0180-4af5-11e9-9065-6a4a7bd51a07.png)

**Comment by jameshupp** [comment]
Gonna deal with the other content issues here momentarily, but

1. Yes. Do this.

2. Yes. Do this, in whatever way #1321 lands.

3. This is trickier. I definitely want to pull a big blue `Save` button away from the nav, especially if we do item 2 from your thinking. We can't do 2 without doing this one. My question is: do we have a secondary button type from the CMS design system we could use? Without any other primary actions on the page, when I see `Save` like this, I think it's the primary action. Pulling it away from the nav helps, but it still feels pretty strong. If nothing else, I'd maybe want to add something to clarify it's in progress. `Save work in progress` or `Save and keep working` or something. Those are too long for a button, that's top of my head stuff there. Could also have a container with a caption in it for the button or something? I dunno.

**Comment by jameshupp** [comment]
Broader thinking here:

* The `Overview` link that some of the sections have is really weird. There's no there there in the content after someone clicks it. I understand the reason; if a top-level nav item has things underneath it, there's no way to click a link and go all the way to the section without selecting a second-level nav and going into a sub-section. But especially in a sticky nav environment, seems like we could/should get rid of it. Where they are in the nav will be right there; if they want to scroll up slightly to the top of that section, it'll be very evident that it's nearby.

* @NAretakis You mentioned this to me the other day and we could use this to expand on the idea a bit (it came up in testing so I'm already ready to jump on it): can you summarize your thinking on moving the Key Personnel out of the top section? Seems like it'll be easier to keep that nav item brief—twice!—if we can get clarity on breaking it apart.

* There are a few places where we're using words these users probably don't need. They know what FFP is, right? And they know that certain things are quarterly?

Bring it all together and I think we can make it a lot shorter, but to maintain clarity we probably need to capture three things for each section header: a nav name, a real name, and the info that an introductory paragraph needs to clarify about what's there. The first two—nav name and real name—probably need a place in the YAML (and I guess then probably the code) but it's very hard to use a short title for the body content _or_ a descriptive title for the nav content while maintaining clarity for users and a fully visible sticky nav.

So if we're good with this, I can put together a doc with that information and @hbillings you and I can go through and see what is workable and not and what needs more clarity from moi.

**Comment by jenniferthibault** [comment]
SUCH GREAT THOUGHTS JAMES! Thank you for going deep on this, it really benefits from a content perspective.

**Button styles**
Yes we sure do have another button style! This one is actually the "default" style, and the blue one was indeed the call to action style. For language, try `Save` since it's shortest, and test it before changing? Other things that come to mind are `Save progress` ? :

![1_State Profile Contact](https://user-images.githubusercontent.com/11636908/54754778-bc7fc080-4bba-11e9-8aad-0620ac77ce4d.png)

**"Overview" language**
Yeah, agreed not ideal. We made that tradeoff to get the vertical nav styles implemented, but the CMS Design System component not allowing links in the parent item is a real stinker for this. Would be excited to see if you think we can avoid it

**Comment by austinhernandez** [comment]
Added thoughts on this in #1308

**Comment by jameshupp** [comment]
Okay, I couldn't figure out how to artfully show my proposed change of header/nav copy in the issue, so I created 🔒 [a Google doc for the nav labeling](https://docs.google.com/document/d/1Ai55bwA0wHctXleuuPF_4tjXklYfFjzVJGvFEhH-090/edit). It's got a table — current vs proposed name for each item — and I know how we love tables. The table is also not artful, but at least the side-by-side is clear!

Most of the following is in the doc, but I want it here for easier discussion.

--
In a couple cases, I worry the new shorter titles might remove too much information to be clear in all contexts (though I don’t think so). If @jeromeleecms or @NAretakis says it _does_ lose clarity or accuracy, we’ll need something to match the short nav name to a longer name in the body of the app itself. 

In that sort of acceptable mismatch scenario, we'd also need to figure out how to handle a two-part label for each section in the code: a nav heading and a main heading. **If this is not an issue** I think I can just change the titles in YAML, but @mgwalker lemme know if there's something else I need to do.

The space-saving results of the proposal, at 100% resolution:

* From a minimum of 25 items when everything is expanded to a minimum of 20. (I continue my war on `Overview` anchors.)
* From three 3-line nav items to zero.
* From six 2-line nav items to … also six. But that’s because we reduced those 3-line items! The current 2-line items go from six to three.

**Comment by jameshupp** [comment]
So where that leaves us: we have a notion of shorter copy. @austinhernandez's work in 1308 gives us a direction for moving the Save button out. And Jen's earlier idea of an Export section moves the PDF button out. I think that gets us something streamlined enough to do the stickifying! Once there is copy signoff — and we definitely need that explicitly — I think the rest of this belongs to @hbillings.

**Comment by jeromeleecms** [comment]
@jameshupp Took a look at the nav labeling + some language suggestions. My only potential concern here is the APD language is based on the OMB approved template, so when we export it, does our language have to match the existing template to comply? Not exactly sure where to start with this -- is this a PRA compliance issue?

**Comment by jeromeleecms** [comment]
@jameshupp Had a brief exchange with @NAretakis. We are both under the impression that the OMB template does not affect the required export language for the PDF, so whatever we settle on seems good to go.

**Comment by hbillings** [comment]
@jameshupp This looks good to me! I'm ready for this to come back on my plate.

**Comment by jameshupp** [comment]
Cool. I think the nav headings are right enough to get started with now. @hbillings should I just go change them in the yaml? If you're trying to implement and things still don't fit acceptably, we can get together with N&J to see if there's more room to maneuver. Not gonna overthink the content at this stage.

**Comment by hbillings** [comment]
@jameshupp That sounds good to me! I'm working on the sticky bits while we speak.
